### PR TITLE
Add support for update ruleset bundles

### PR DESCRIPTION
**Pull request by cfir**
- add updateRuleset.py for implementations
- new methods for the SDK:
--static method getJson() to load json
-- getAccountBundles() return account bumdles
--updateRuleBundleByID() update ruleset bundle by ID

### CODE REVIEW

File: dome9ApiV2Py/dome9ApiV2Py.py [hunk 1]
```diff
@@ -8,7 +8,15 @@
 class Dome9ApiSDK(object):
 	REGION_PROTECTION_MODES = ['FullManage', 'ReadOnly', 'Reset']
 	SEC_GRP_PROTECTION_MODES = ['FullManage', 'ReadOnly']
-	
+
+	@staticmethod
+	def getJson(path):
+		try:
+			with open(path) as jsonFile:
```
Comment (uditamar) [inline dome9ApiV2Py/dome9ApiV2Py.py:15]: bad indentation

File: dome9ApiV2Py/dome9ApiV2Py.py [hunk 2]
```diff
@@ -8,7 +8,15 @@
...
+				return json.load(jsonFile)
+		except Exception as err:
```
Comment (uditamar) [inline dome9ApiV2Py/dome9ApiV2Py.py:17]: search for a proper exception

File: dome9ApiV2Py/dome9ApiV2Py.py [hunk 3]
```diff
@@ -8,7 +8,15 @@
...
+			exit(err)
```
Comment (uditamar) [inline dome9ApiV2Py/dome9ApiV2Py.py:18]: raies - no exit

File: dome9ApiV2Py/dome9ApiV2Py.py [hunk 4]
```diff
@@ -284,7 +292,27 @@ def runAssessmenBundle(self, assReq, outAsJson=False): # assessmentRequest
...
+	def getAccountBundles(self, outAsJson=False):
+		apiCall = self.get(route='CompliancePolicy')
+
+		if outAsJson:
+			print(json.dumps(apiCall))
+		return apiCall
+
+	def updateRuleBundleByID(self, ruleID, ruleSet, outAsJson=False):
+
+		data = {
```
Comment (uditamar) [inline dome9ApiV2Py/dome9ApiV2Py.py:305]: Mixed quotes 
bad indentation

File: implementation/updateRuleset.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+		print ('bundle result is\n{}'.format(json.dumps(bundleResult)))
+
+if __name__ == '__main__':
+	parser = argparse.ArgumentParser(description='')
+	useExample = '--dome9ApiKeyID 111111-2222-3333-3333-3333333 --dome9ApiKeySecret qwerrtyy --rulesetJsonPath ruleset.json --bundleID 11111 --cloudAccountID 11111111111'
+
+	parser.epilog = 'Example of use: {} {}'.format(__file__, useExample)
+	parser.add_argument('--dome9ApiKeyID',          required=True,  type=str, help='(required) Dome9 Api key')
+	parser.add_argument('--dome9ApiKeySecret',      required=True,  type=str, help='(required) Dome9 secret key')
+	parser.add_argument('--bundleID',      required=True,  type=str, help='(required) Dome9 bundle ID')
```
Comment (uditamar) [inline implementation/updateRuleset.py:36]: align

File: implementation/updateRuleset.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import uuid
+import json
+from dome9ApiV2Py import Dome9ApiClient
+
+
+class UpdateRuleset(object):
+
+	def __init__(self, args):
+		self.args = args
+		self.d9client = Dome9ApiClient(apiKeyID=args.dome9ApiKeyID, apiSecret=args.dome9ApiKeySecret)
```
Comment (uditamar) [inline implementation/updateRuleset.py:12]: d9Client

File: implementation/updateRuleset.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
```
Comment (uditamar) [inline implementation/updateRuleset.py:13]: extra space

File: implementation/updateRuleset.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+	def mainProcess(self):
+		print ('loading bundle from {}'.format(self.args.rulesetJsonPath))
+		ruleBundleObject = self.d9client.getJson(self.args.rulesetJsonPath)
+		print ('Updating bundle {}'.format(self.args.bundleID))
+		self.d9client.updateRuleBundleByID(self.args.bundleID, ruleBundleObject)
+		bundle = {
```
Comment (uditamar) [inline implementation/updateRuleset.py:20]: should be at the ctor

File: implementation/updateRuleset.py [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+		print ('Running assessment bundle')
+		bundleResult  = self.d9client.runAssessmenBundle(bundle)
+		print ('bundle result is\n{}'.format(json.dumps(bundleResult)))
+
+if __name__ == '__main__':
+	parser = argparse.ArgumentParser(description='')
+	useExample = '--dome9ApiKeyID 111111-2222-3333-3333-3333333 --dome9ApiKeySecret qwerrtyy --rulesetJsonPath ruleset.json --bundleID 11111 --cloudAccountID 11111111111'
+
+	parser.epilog = 'Example of use: {} {}'.format(__file__, useExample)
+	parser.add_argument('--dome9ApiKeyID',          required=True,  type=str, help='(required) Dome9 Api key')
```
Comment (uditamar) [inline implementation/updateRuleset.py:34]: drop all (required) we have arg parse help for it

File: implementation/updateRuleset.py [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+		print ('loading bundle from {}'.format(self.args.rulesetJsonPath))
+		ruleBundleObject = self.d9client.getJson(self.args.rulesetJsonPath)
+		print ('Updating bundle {}'.format(self.args.bundleID))
+		self.d9client.updateRuleBundleByID(self.args.bundleID, ruleBundleObject)
+		bundle = {
+			'id': self.args.bundleID,
+			'cloudAccountId': self.args.cloudAccountID,
+			'requestId': str(uuid.uuid4())
+		}
+		print ('Running assessment bundle')
```
Comment (uditamar) [inline implementation/updateRuleset.py:25]: add the requestId to the print

File: implementation/updateRuleset.py [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+		bundleResult  = self.d9client.runAssessmenBundle(bundle)
+		print ('bundle result is\n{}'.format(json.dumps(bundleResult)))
```
Comment (uditamar) [inline implementation/updateRuleset.py:27]: very risky

File: implementation/updateRuleset.py [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+	parser.epilog = 'Example of use: {} {}'.format(__file__, useExample)
+	parser.add_argument('--dome9ApiKeyID',          required=True,  type=str, help='(required) Dome9 Api key')
+	parser.add_argument('--dome9ApiKeySecret',      required=True,  type=str, help='(required) Dome9 secret key')
+	parser.add_argument('--bundleID',      required=True,  type=str, help='(required) Dome9 bundle ID')
+	parser.add_argument('--rulesetJsonPath',      required=True,  type=str, help='(required) Location for ruleset json file')
+	parser.add_argument('--cloudAccountID',      required=True,  type=str, help='(required) Dome9 external account ID')
+
+	arguments = parser.parse_args()
+	TestD9Api = UpdateRuleset(arguments)
+	TestD9Api.mainProcess()
```
Comment (uditamar) [inline implementation/updateRuleset.py:42]: missing eof nl

Comment (uditamar) [inline implementation/updateRuleset.py:42]: missing eof nl

File: implementation/updateRuleset.py [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+
+	parser.epilog = 'Example of use: {} {}'.format(__file__, useExample)
+	parser.add_argument('--dome9ApiKeyID',          required=True,  type=str, help='(required) Dome9 Api key')
+	parser.add_argument('--dome9ApiKeySecret',      required=True,  type=str, help='(required) Dome9 secret key')
+	parser.add_argument('--bundleID',      required=True,  type=str, help='(required) Dome9 bundle ID')
+	parser.add_argument('--rulesetJsonPath',      required=True,  type=str, help='(required) Location for ruleset json file')
+	parser.add_argument('--cloudAccountID',      required=True,  type=str, help='(required) Dome9 external account ID')
+
+	arguments = parser.parse_args()
+	TestD9Api = UpdateRuleset(arguments)
```
Comment (uditamar) [inline implementation/updateRuleset.py:41]: vars are lowered case
bad name

File: implementation/ruleset.json [hunk 14]
```diff
@@ -0,0 +1,3 @@
+[
```
Comment (uditamar) [inline implementation/ruleset.json:1]: explain

File: implementation/updateRuleset.py (hunk 12 above)
Comment (uditamar) [inline implementation/updateRuleset.py:42]: missing eof nl

File: implementation/updateRuleset.py [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+			'id': self.args.bundleID,
+			'cloudAccountId': self.args.cloudAccountID,
+			'requestId': str(uuid.uuid4())
+		}
+		print ('Running assessment bundle')
+		bundleResult  = self.d9client.runAssessmenBundle(bundle)
+		print ('bundle result is\n{}'.format(json.dumps(bundleResult)))
+
+if __name__ == '__main__':
+	parser = argparse.ArgumentParser(description='')
```
Comment (uditamar) [inline implementation/updateRuleset.py:30]: missing

File: implementation/updateRuleset.py (hunk 13 above)
Comment (uditamar) [inline implementation/updateRuleset.py:41]: TestD9Api: bad name , vars are lowered case

**Review by uditamar — CHANGES_REQUESTED** [review]

File: implementation/ruleset.json (hunk 14 above)
Comment (paveli-d9) [reply implementation/ruleset.json:1]: Discussed with Cfir, should remain as is.

**Pull request closed without merging**